vm/systemSmartContracts/staking.go

```diff
@@ -311,6 +311,10 @@ func (r *stakingSC) unJail(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		r.eei.AddReturnMessage("cannot unJail a key that is not registered")
 		return vmcommon.UserError
 	}
+	if !stakedData.Jailed && !r.eei.CanUnJail(args.Arguments[0]) {
```

Comment:
is this verification correct? if not jailed, why try "CanUnJail"?

Reply:
because there might be slight differences between stakingSC and validator statistics. at end of epoch not all with BAD rating are jailed - those still selected (as they are needed) might turn on their PC and have their rating back to normal values again